pkg/operator/controller/dns/controller.go

```diff
@@ -292,3 +340,141 @@ func dnsZoneStatusSlicesEqual(a, b []iov1.DNSZoneStatus) bool {
 
 	return true
 }
+
+// ToDNSRecords returns reconciliation requests for all DNSRecords.
+func (r *reconciler) ToDNSRecords(o handler.MapObject) []reconcile.Request {
+	var requests []reconcile.Request
+	records := &operatoringressv1.DNSRecordList{}
+	if err := r.cache.List(context.Background(), records, client.InNamespace(r.Namespace)); err != nil {
+		log.Error(err, "failed to list dnsrecords", "related", o.Meta.GetSelfLink())
+		return requests
+	}
+	for _, record := range records.Items {
+		log.Info("queueing dnsrecord", "name", record.Name, "related", o.Meta.GetSelfLink())
+		request := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: record.Namespace,
+				Name:      record.Name,
+			},
+		}
+		requests = append(requests, request)
+	}
+	return requests
+}
+
+// createDNSProvider creates a DNS manager compatible with the given cluster
+// configuration.
+func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *configv1.PlatformStatus) (dns.Provider, error) {
+	// If no DNS configuration is provided, don't try to set up provider clients.
+	// TODO: the provider configuration can be refactored into the provider
```

Comment:
Does it make sense to work on this TODO now, since the DNS controller is already being moved around? If you think this would be helpful at the moment, I could work on it.

Reply:
Well, I guess it would make sense to wait til #416 is finished to make things less confusing, but that's fine.

Reply:
> Does it make sense to work on this TODO now, since the DNS controller is already being moved around? 

I prefer that the TODO is addressed in a separate PR due to time constraints and to keep this PR focused on the refactor needed for supporting AWS/Azure custom endpoints.